quipucords/fingerprinter/task.py

```diff
@@ -235,15 +237,23 @@ def _process_details_report(self, manager_interrupt, details_report):
             fingerprint_dict['deployment_report'] = deployment_report.id
             serializer = SystemFingerprintSerializer(data=fingerprint_dict)
             if serializer.is_valid():
-                number_valid += 1
-                serializer.save()
-                fingerprint_dict.pop('deployment_report')
-                # Serialize the date
-                for field in SystemFingerprint.DATE_FIELDS:
-                    if fingerprint_dict.get(field, None):
-                        fingerprint_dict[field] = date_field.to_representation(
-                            fingerprint_dict.get(field))
-                final_fingerprint_list.append(fingerprint_dict)
+                try:
+                    serializer.save()
+                    fingerprint_dict.pop('deployment_report')
+                    # Serialize the date
+                    for field in SystemFingerprint.DATE_FIELDS:
+                        if fingerprint_dict.get(field, None):
+                            fingerprint_dict[field] = \
+                                date_field.to_representation(
+                                    fingerprint_dict.get(field))
+                    final_fingerprint_list.append(fingerprint_dict)
+                    number_valid += 1
+                except DataError as error:
+                    number_invalid += 1
+                    self.scan_task.log_message(
+                        'The following fingerprint failed with error "%s": %s'
```

Comment:
How about this?  `The fingerprint could not be saved.  Fingerprint: %s.  Error: %s`